dandischema/metadata.py

```diff
@@ -76,15 +76,21 @@ def publish_model_schemata(releasedir: str) -> Path:
     version = models.get_schema_version()
     vdir = Path(releasedir, version)
     vdir.mkdir(exist_ok=True, parents=True)
-    (vdir / "dandiset.json").write_text(models.Dandiset.schema_json(indent=2))
-    (vdir / "asset.json").write_text(models.Asset.schema_json(indent=2))
+    (vdir / "dandiset.json").write_text(
+        models.Dandiset.schema_json(indent=2).rstrip("\n") + "\n"
```

Comment:
absent trailing new line was an issue?

we don't care about windows here, or python would magically convert line breaks to windows ones?

I would have probably came up with a helper `ensure_trailing_space` or alike to avoid duplication.

Reply:
or better, as I have suggested before, I think we should overload `schema_json` and centralize all our tuneups (indent and trailing line) in there

Reply:
python magically converts or rather both `open`/`write_text` does as far as i could tell.

Reply:
i'm still a little hesitant on overwriting `schema_json` as then we would need to ensure that everything we have is derived from a common `BaseModel` that just overrides `schema_json`. doable, but hesitant.

Reply:
isn't everything relevant here is a subclass of `DandiBaseModel` or otherwise we would have other issues (since `DandiBaseModel` does already to_dictrepr and unvalidated and we would fail if it didn't?).  Anyways -- could be harmonized after this PR.  But the question just keeps coming, so we better deal with it.... I guess next time ;)